Build does not work with gold

```

[100%] Linking CXX shared module _C.cpython-38-x86_64-linux-gnu.so
/private/home/ezyang/bin/ld: undefs: unknown -z option
/private/home/ezyang/bin/ld: use the --help option for usage information
collect2: error: ld returned 1 exit status
make[2]: *** [src/python/torchdistx/_C/CMakeFiles/torchdistx-py.dir/build.make:140: src/python/torchdistx/_C/_C.cpython-38-x86_64-linux-gnu.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:914: src/python/torchdistx/_C/CMakeFiles/torchdistx-py.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
(/scratch/ezyang/pytorch-scratch2-env) ezyang
```

--- a/cmake/Helpers.cmake
+++ b/cmake/Helpers.cmake
@@ -227,12 +227,7 @@ function(torchdist_add_target target)
                 LINKER:-z,relro
         )
 
-        if(arg_PYTHON_MODULE)
-            target_link_options(${target}
-                PRIVATE
-                    LINKER:-z,undefs
-            )
-        else()
+        if(NOT arg_PYTHON_MODULE)
             target_link_options(${target}
                 PRIVATE
                     LINKER:-z,defs

